pkg/app/piped/cmd/piped/piped.go

```diff
@@ -487,6 +491,57 @@ func (p *piped) run(ctx context.Context, input cli.Input) (runErr error) {
 		})
 	}
 
+	// Check for stop command.
+	{
+		stopCommandHandler := func(ctx context.Context) (bool, error) {
+			input.Logger.Debug("fetch unhandled piped commands")
+
+			commands := commandLister.ListPipedCommands()
+			if len(commands) == 0 {
+				return false, nil
+			}
+
+			var stopCmd *model.ReportableCommand
+			for _, command := range commands {
+				if command.IsRestartPipedCmd() {
+					stopCmd = &command
+					break
+				}
+			}
+			if stopCmd == nil {
+				return false, nil
+			}
+
+			if err := stopCmd.Report(ctx, model.CommandStatus_COMMAND_SUCCEEDED, nil, []byte(stopCmd.Id)); err != nil {
+				return false, fmt.Errorf("failed to report command %s: %w", stopCmd.Id, err)
+			}
+
+			return true, nil
+		}
+
+		group.Go(func() error {
+			input.Logger.Info("start running piped stop checker")
+			ticker := time.NewTicker(p.gracePeriod)
+			for {
+				select {
+				case <-ticker.C:
+					shouldStop, err := stopCommandHandler(ctx)
+					// Don't return an error to continue this goroutine execution.
+					if err != nil {
+						input.Logger.Error("failed to check/handle piped stop command", zap.Error(err))
+					}
+					if shouldStop {
+						input.Logger.Info("stop piped due to restart piped requested")
+						cancel()
```

Comment:
This sound unnecessary. 👀

Reply:
oops, I misunderstood and this is required, so let's keep this.

Reply:
Sorry, I did not catch your point around this.
If `shouldStop` is true, this run method returns an error, right?

Reply:
sorry which run method do you say?

Reply:
OK, I totally understand this after talking directly. Thanks